wallet/src/main/java/io/neow3j/wallet/Nep5.java

```diff
@@ -0,0 +1,75 @@
+package io.neow3j.wallet;
+
+import io.neow3j.contract.ContractParameter;
+import io.neow3j.contract.ScriptHash;
+import io.neow3j.protocol.Neow3j;
+import io.neow3j.protocol.core.methods.response.NeoInvoke;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+
+public class Nep5 {
+
+    private Neow3j neow3j;
+    private ScriptHash fromContractScriptHash;
+
+    public Nep5() {
+    }
+
+    private Nep5(Builder builder) {
+        this.neow3j = builder.neow3j;
+        this.fromContractScriptHash = builder.fromContractScriptHash;
+    }
+
+    public BigInteger totalSupply() throws IOException {
+        List<ContractParameter> params = Arrays.asList(
+                ContractParameter.string("totalSupply"),
+                ContractParameter.array());
+        NeoInvoke send = neow3j.invoke(this.fromContractScriptHash.toString(), params).send();
+        return send.getInvocationResult().getStack().get(1).asByteArray().getAsNumber();
+    }
+
+    public String name() throws IOException {
+        List<ContractParameter> params = Arrays.asList(
+                ContractParameter.string("name"),
+                ContractParameter.array());
+        NeoInvoke send = neow3j.invoke(this.fromContractScriptHash.toString(), params).send();
+        return send.getInvocationResult().getStack().get(1).asByteArray().getAsString();
+    }
+
+    public String symbol() throws IOException {
+        List<ContractParameter> params = Arrays.asList(
+                ContractParameter.string("symbol"),
+                ContractParameter.array());
+        NeoInvoke send = neow3j.invoke(this.fromContractScriptHash.toString(), params).send();
+        return send.getInvocationResult().getStack().get(1).asByteArray().getAsString();
+    }
+
+    public BigInteger decimals() throws IOException {
+        List<ContractParameter> params = Arrays.asList(
+                ContractParameter.string("decimals"),
+                ContractParameter.array());
+        NeoInvoke send = neow3j.invoke(this.fromContractScriptHash.toString(), params).send();
+        return (BigInteger) send.getInvocationResult().getStack().get(1).getValue();
+    }
+
+    public static class Builder {
+        private Neow3j neow3j;
+        private ScriptHash fromContractScriptHash;
+        public Builder(Neow3j neow3j) {
+            this.neow3j = neow3j;
+        }
+
+        public Nep5.Builder fromContract(ScriptHash contractScriptHash) {
```

Comment:
@Priyanshu-soni could we, somehow, check whether this contract is a Nep5 standard using the JSON-RPC `invoke`?
@csmuller what's your opinion? Should we do this check here? Or an exception would be thrown anyway once a method that does not exist is called in this contract?

Reply:
I don't think we should check the contract for every NEP-5 method. With the possibilities in Neo 2 this would imply doing `invoke`s for every method.
Though, there is a new standard, NEP-10, which defines that a smart contract should have a method `supportedStandards` that would tell us that the contract supports the NEP-5 API.
For now we can ignore that and leave it to the developer to decide if the wrapped contract fulfills the NEP-5 standard. I will keep track of NEP-10.